src/codebase.md

```diff
@@ -260,3 +260,41 @@ crates optimally for your semantic needs.
 | [ring](https://docs.rs/ring/)           | the leading crypto library         |
 | [nalgebra](https://docs.rs/nalgebra/)   | linear algebra                     |
 | [once_cell](https://docs.rs/once_cell/) | complex static data                |
+
+## How should I call C++ functions from Rust and vice-versa?
+
+Behind the scenes, Rust natively supports calling C functions using `extern "C"`,
+`#[repr(C)]` and `#[no_mangle]`. This is, of course, error prone and involves
+lots of `unsafe` code. So don't do that: there are better ways.
+
+Your initial decision should be a matter of policy. Choose between:
+
+* I want to manually specify the cross-language interface in order to keep
+  a degree of isolation and to avoid proliferation of uncontrolled
+  cross-language boundaries. Or:
+* I want to generate comprehensive bindings automatically from existing C++
+  headers to allow arbitrary calls from Rust to C++.
+
+If you choose the former approach, the modern standard for C++ interoperability
+is [cxx](https://cxx.rs). You specify cross-language interfaces in a
+Rust-like interface definition language (within your Rust file). Both Rust and
+C++ bindings code is generated, to make the interface natural in each language. For
+example, you'll find idiomatic Rust wrappers for [`std::string`](https://docs.rs/cxx/1.0.50/cxx/struct.CxxString.html) and [`std::unique_ptr`](https://docs.rs/cxx/1.0.50/cxx/struct.UniquePtr.html).
+This doesn't just make things easier and more idiomatic: by understanding
+standard C++ ownership models, `cxx` can allow Rust code to interact with
+C++ without widespread use of `unsafe`.
+
+The latter approach doesn't work as smoothly, tempting as it may seem.
```

Comment:
When reading this, I initially thought "latter approach" refers back to the previous paragraph, not the bullet point a way up.

Maybe refer to the "first" and "second" approach instead of "former" and "latter" -- and maybe even repeat what the second approach is? E.g.

"The second approach -- generating bindings automatically from C++ headers -- doesn't work as smoothly, tempting as it may seem."

Reply:
Done